optuna/samplers/_tpe/parzen_estimator.py

```
@@ -191,13 +191,22 @@ def _calculate_categorical_distributions(
             )
 
         n_kernels = len(observations) + parameters.consider_prior
-        assert parameters.prior_weight is not None
-        weights = np.full(
-            shape=(n_kernels, n_choices),
-            fill_value=parameters.prior_weight / n_kernels,
-        )
+        if parameters.consider_prior:
+            assert parameters.prior_weight is not None
+            weights = np.full(
+                shape=(n_kernels, n_choices),
+                fill_value=parameters.prior_weight / n_kernels,
+            )
+        else:
+            weights = np.full(
+                shape=(n_kernels, n_choices),
+                fill_value=0.0,
+            )
         observed_indices = observations.astype(int)
-        if param_name in parameters.categorical_distance_func:
+        if (
+            param_name in parameters.categorical_distance_func
+            and parameters.prior_weight is not None
```

[Question] Why do we need this condition?
More specifically, isn't it better to raise an error when we get `prior_weight = None` and `categorical_distance_func` simultaneously?

---

I realized that my previous implementation might not be entirely correct. There could be cases where `categorical_distance_func` is used even when `consider_prior` is False, which means `prior_weight` could be None. To address this, I change the logic to set `coef` to 1.0 when `consider_prior` is False. I've updated the logic accordingly—could you take a look and let me know if there's anything I misunderstood?